src/ucp/api/ucp.h

```diff
@@ -1207,6 +1208,24 @@ typedef struct ucp_mem_map_params {
       * UCP_MEM_MAP_PROT_REMOTE_READ|UCP_MEM_MAP_PROT_REMOTE_WRITE.
       */
      unsigned               prot;
+
+     /*
+      * Memory type (for possible memory types see @ref ucs_memory_type_t)
+      * This value is optional. The meaning of this field depends on the
+      * operation type.
+      *
+      * - Memory allocation: (@ref UCP_MEM_MAP_ALLOCATE flag is set) This field
+      *    specifies the type of memory to allocate. If it's not set (along with its
+      *    corresponding bit in the field_mask - @ref UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE),
+      *    @ref UCS_MEMORY_TYPE_HOST will be assumed by default.
+      *
+      * - Memory registration: This field specifies the type of memory which is
+      *    pointed by @ref ucp_mem_map_params.address. If it's not set (along with its
+      *    corresponding bit in the field_mask - @ref UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE),
+      *    or set to @ref UCS_MEMORY_TYPE_UNKNOWN, the memory type will be dectected
+      *    internally.
```

Comment:
I suggest to add that `it is advised to users to set memory type to accelerate in order to reduce the cost of automatic memory type detection.`

Reply:
@shamisp  can you check the new text?

Reply:
@bureddy Looks for for me. 👍